resample_states_verification/NOTES.md

```diff
@@ -0,0 +1,58 @@
+
+Testing of changes to use mock GET_PITCH commands
+=================================================
+
+Used the timelines code to make a new test database and cmd states for 2010.  This made
+the archive by stepping through the days, which we don't care about for this, but, in the
+timelines repo:
+
+
+    import timelines_test
+    timelines_test.all_2010(cmd_state_ska='/data/fido/ska')
+
+I then used fetch_states against the flight cmd_states.h5 and the testing cmd_states.h5 to
+create dat files limited to the columns that really should not be changing due to this
+update.
+
+    from Chandra.cmd_states.get_cmd_states import fetch_states
+    from astropy.table import Table
+    
+    official = fetch_states(start='2010:005', stop='2010:360', vals=['obsid', 'power_cmd',
+      'si_mode', 'pcad_mode', 'vid_board', 'clocking', 'fep_count', 'ccd_count', 'simpos',
+      'simfa_pos', 'hetg', 'letg'], server='/proj/sot/ska/data/cmd_states/cmd_states.h5')
+    Table(official).write('official_minus_point.dat', format='ascii')
+    
+    test = fetch_states(start='2010:005', stop='2010:360', vals=['obsid', 'power_cmd',
+      'si_mode', 'pcad_mode', 'vid_board', 'clocking', 'fep_count', 'ccd_count', 'simpos',
+      'simfa_pos', 'hetg', 'letg'], server='t/all_2010/cmd_states.h5')
+    Table(test).write('test_minus_point.dat', format='ascii')
+    
+    meld official_minus_point.dat test_minus_point.dat
+
+The small set of small timing diffs are acceptable on the non-pointing-related states.
```

Comment:
Related to changing the interval to 10000s, is there anything else we
should do to try to make the generated pitch commands end up at the same
times over multiple cmd_state updates?

Because these pitch samples are ending up at times based on the start time
for update_cmd_states and the range of cmds, we're going to get different
times every run of the tool, and that will make for more overall
delete/reinserts.  Maybe if these are just at tstarts that are rounded to
the nearest 10k or the like, we'd have fewer updates.

Somewhat a premature optimization, but perhaps also helpful for our own
logging (as it would keep us from seeing changes and updates where there
aren't any).

On Fri, Jun 3, 2016 at 3:03 PM, Jean Connelly <jeanconn@head.cfa.harvard.edu
> wrote:

> OK.  Thanks!
>
>
> On Fri, Jun 3, 2016 at 2:57 PM, Tom Aldcroft <notifications@github.com>
> wrote:
>
>> In resample_states_verification/NOTES.md
>> <https://github.com/sot/cmd_states/pull/21#discussion_r65758887>:
>>
>> > +    from Chandra.cmd_states.get_cmd_states import fetch_states
>> > +    from astropy.table import Table
>> > +
>> > +    official = fetch_states(start='2010:005', stop='2010:360', vals=['obsid', 'power_cmd',
>> > +      'si_mode', 'pcad_mode', 'vid_board', 'clocking', 'fep_count', 'ccd_count', 'simpos',
>> > +      'simfa_pos', 'hetg', 'letg'], server='/proj/sot/ska/data/cmd_states/cmd_states.h5')
>> > +    Table(official).write('official_minus_point.dat', format='ascii')
>> > +
>> > +    test = fetch_states(start='2010:005', stop='2010:360', vals=['obsid', 'power_cmd',
>> > +      'si_mode', 'pcad_mode', 'vid_board', 'clocking', 'fep_count', 'ccd_count', 'simpos',
>> > +      'simfa_pos', 'hetg', 'letg'], server='t/all_2010/cmd_states.h5')
>> > +    Table(test).write('test_minus_point.dat', format='ascii')
>> > +
>> > +    meld official_minus_point.dat test_minus_point.dat
>> > +
>> > +The small set of small timing diffs are acceptable on the non-pointing-related states.
>>
>> Do we have a place to put test outputs that makes some sense?
>>
>> No, but I don't think they really need to have a persistent place outside
>> of your own dev repo. Given that you have nicely documented how to create
>> the files then anyone could do that on demand.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/sot/cmd_states/pull/21/files/2f26e493c9e2aaec9bb34765ee17d1cf8a022e08#r65758887>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AAwT1HTHYiVGrFrq2qsSFKvOsrRaCeMsks5qIHkPgaJpZM4F7T73>
>> .
>>
>
>